rq/worker.py

```diff
@@ -401,12 +401,12 @@ def refresh(self):
 
     @property
     def should_run_maintenance_tasks(self):
-        """Maintenance tasks should run on first startup or every 10 minutes."""
-        if self.last_cleaned_at is None:
-            return True
-        if (utcnow() - self.last_cleaned_at) > timedelta(seconds=self.maintenance_interval):
-            return True
-        return False
+        """
+        Maintenance tasks should run on first startup or every maintenance
+        interval (defaults to 10 minutes).
+        """
+        l_c_at = self.last_cleaned_at = self.last_cleaned_at or worker_registration.get_last_cleaned_at(self.connection)
```

Comment:
If we're storing the last cleaning time at `self.last_cleaned_at`, we don't need `l_c_at` variable.

Reply:
l_c_at is there just for having a shorter line below